relayer/relayer/src/handlers/eth.rs

```diff
@@ -91,39 +89,26 @@ impl EthereumEventHandler {
             let amount = amount.as_u128();
             let request_nonce = request_nonce.as_u128();
 
-            while contract
-                .needs_signature(
-                    azero_connection.as_connection(),
+            contract
```

Comment:
looping is indeed probably not needed here, but the guardian could still confirm if his signature is needed before signing the request

Reply:
He cannot because we cannot make this call for the last finalized block, so we could omit a tx that will actually need our signature.